src/MemoryTracing/SystemMemoryInfoProducer.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryTracing/SystemMemoryInfoProducer.h"
+
+#include <absl/synchronization/mutex.h>
+#include <absl/time/time.h>
+
+#include "MemoryTracingUtils.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+
+namespace orbit_memory_tracing {
+
+using orbit_grpc_protos::SystemMemoryUsage;
+
+void SystemMemoryInfoProducer::Run() {
+  orbit_base::SetCurrentThreadName("SysMemPr::Run");
+
+  CHECK(listener_ != nullptr);
+  ProduceSystemMemoryUsageAndSendToListener();
+}
+
+void SystemMemoryInfoProducer::ProduceSystemMemoryUsageAndSendToListener() {
+  absl::Time scheduled_time = absl::Now();
+  absl::MutexLock lock(&exit_requested_mutex_);
+  while (!exit_requested_) {
+    ErrorMessageOr<SystemMemoryUsage> system_memory_usage = GetSystemMemoryUsage();
+    if (system_memory_usage.has_value()) {
```

Comment:
Again what happens to the error case? I feel that at least they should be logged.

Reply:
Thanks for pointing out this. The same with `GetCGroupMemoryUsage`, this method returns error message if there are failures for reading the relevant linux files into string or extracting desired values from the file content. The error message is logged out in the  `GetSystemMemoryUsage` method.